docs/Operators.md

```diff
@@ -7630,39 +7826,79 @@ This version of the operator has been available since version 16 of the default
+
+<dt><tt>grid</tt> (non-differentiable) : T2</dt>
+<dd>4-D tensor of shape (N, H_out, W_out, 2), where H_out and W_out are the height and width of grid and output. This is the tensor that GridSample uses to determine where to do lookups in input. This parameter is allowed to be integer, but this would typically indicate a mistake as this can only access the input data at its corners. It is not allowed for N, H_out or W_out to be zero.</dd>
 </dl>
 
 #### Outputs
 
 <dl>
-<dt><tt>Y</tt> (differentiable) : T2</dt>
+<dt><tt>output</tt> (differentiable) : T1</dt>
 <dd>4-D tensor of shape (N, C, H_out, W_out).</dd>
 </dl>
 
 #### Type Constraints
 
 <dl>
-<dt><tt>T1</tt> : tensor(uint8), tensor(uint16), tensor(uint32), tensor(uint64), tensor(int8), tensor(int16), tensor(int32), tensor(int64), tensor(float16), tensor(float), tensor(double), tensor(string), tensor(bool), tensor(complex64), tensor(complex128)</dt>
-<dd>Constrain input types to all tensor types.</dd>
+<dt><tt>T1</tt> : tensor(float16), tensor(float), tensor(double), tensor(complex64), tensor(complex128)</dt>
+<dd>Constrain input and output types to float and complex tensors.</dd>
 <dt><tt>T2</tt> : tensor(float16), tensor(float), tensor(double)</dt>
-<dd>Constrain output types to float tensors.</dd>
+<dd>Constrain grid operand type to float tensors.</dd>
 </dl>
 
+#### Unresolved
+
+**Some concerns from the issue reported on GridSample that are not resolved here:
+
+1. We need dimensions higher than 4D for GridSample. Can the spec run
+ahead of the Microsoft ONNX implementation or does it need to wait for
+implementation to be done first?
+
+2. Math is not displaying properly on Github. Should it? How should
+math be typeset properly in the ONNX spec? Presumably we want
+LaTeX-type math since ML has a lot of math in it.
+
+3. For nearest_mode, I did not follow resize(). I removed ceil and
+floor, as I don't understand when those would be used. I added
+round_to_even and put it as the default, as it seems the most balanced
+/ unbiased rounding mode of these ones.
+
+4. I set align_corners as int32, instead of int. Many other ops just
+say "int". Not sure what is correct here. I think the spec should say
+what the type is precisely.
+
+5. I disallowd integers inputs. They make sense mathematically, but
+logically it would almost always be an errro (maybe actually just
+always) to use this, as you'd hit the grid only on the corners, which
+doesn't make a lot of sense and is almost certainly something the user
+didn't intend to do and probably indicates either a bug in the user
+code or a misunderstanding of how the op works. So I think it's better
+to get an error in this case, but let me know what you think.
+
+6. I disallowed zero dimensions. Not clear what the general thinking
+is in ONNX for zero dimensions. Maybe it should be allowed? I would
+normally allow it, but it's hard to interpolate among no information,
+not clear what e.g. the border padding mode should do in this case.
```

Comment:
Hi @broune , basically the spec need to consider the op implementation so we should keep the description in accordance with kernel. Do you need further help and we can complete this change proposal together?

Reply:
I would like to know how you can handle zero dimensions. I think there's no good way to handle them, so I assume it's an error.